[CI] InternalEngineTests.testIndexWriterInfoStream failure

Reproduce line:

```
./gradlew ':server:unitTest' -Dtests.seed=23A79BFD76C74CE -Dtests.class=org.elasticsearch.index.engine.InternalEngineTests -Dtests.method="testIndexWriterInfoStream" -Dtests.security.manager=true -Dtests.locale=zh-Hans-CN -Dtests.timezone=Atlantic/Stanley -Dcompiler.java=12 -Druntime.java=12
```

I was unable to reproduce it locally

Failure:

```
FAILURE 0.36s J13 | InternalEngineTests.testIndexWriterInfoStream <<< FAILURES!
13:55:36    > Throwable #1: java.lang.AssertionError: 
13:55:36    > Expected: an empty collection
13:55:36    >      but: <[Attempted to append to non-started appender testIndexWriterInfoStream]>
13:55:36    > 	at __randomizedtesting.SeedInfo.seed([23A79BFD76C74CE:303742EE3F4AC155]:0)
13:55:36    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
13:55:36    > 	at org.elasticsearch.test.ESTestCase.checkStaticState(ESTestCase.java:516)
13:55:36    > 	at org.elasticsearch.test.ESTestCase.after(ESTestCase.java:365)
13:55:36    > 	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
13:55:36    > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
13:55:36    > 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
13:55:36    > 	at java.base/java.lang.Thread.run(Thread.java:835)
```

Branch: 6.8

Failure happens when running tear down validation in ESTestCase.checkStaticState(...)

Build url: https://gradle-enterprise.elastic.co/s/csltgh2b5h22w

--- a/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
+++ b/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
@@ -2369,7 +2369,7 @@ public class InternalEngineTests extends EngineTestCase {
             engine.index(indexForDoc(doc));
             engine.flush();
             assertTrue(mockAppender.sawIndexWriterMessage);
-
+            engine.close();
         } finally {
             Loggers.removeAppender(rootLogger, mockAppender);
             mockAppender.stop();

